.yarn/patches/@metamask-providers-npm-18.2.0-3616fe3170.patch

```diff
@@ -0,0 +1,188 @@
+diff --git a/dist/BaseProvider.cjs b/dist/BaseProvider.cjs
+index bcb7967af0b6369cc843a159e1c369e5d8a91f12..ad8b431fc9f6c7b4bffb43591ffc5f9e662f1a45 100644
+--- a/dist/BaseProvider.cjs
++++ b/dist/BaseProvider.cjs
+@@ -18,6 +18,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
+ exports.BaseProvider = void 0;
+ const json_rpc_engine_1 = require("@metamask/json-rpc-engine");
+ const rpc_errors_1 = require("@metamask/rpc-errors");
++const { v4: uuidV4 } = require('uuid');
+ const safe_event_emitter_1 = __importDefault(require("@metamask/safe-event-emitter"));
+ const fast_deep_equal_1 = __importDefault(require("fast-deep-equal"));
+ const messages_1 = __importDefault(require("./messages.cjs"));
+@@ -134,9 +135,11 @@ class BaseProvider extends safe_event_emitter_1.default {
+         }
+         const payload = params === undefined || params === null
+             ? {
++                id: uuidV4(),
```

Comment:
An id field was necessary to ensure that requests originating from StreamProvider could be correctly matched with their responses and therefore resolve properly.